includes/prompt-tags/class-prompt-tag.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Newspack Popups Prompt Tag
+ *
+ * @package Newspack
+ */
+
+namespace Newspack\Campaigns;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Prompt Tag class.
+ */
+final class Prompt_Tag {
+	/**
+	 * Tag name
+	 *
+	 * @var string
+	 */
+	public $name;
+
+	/**
+	 * Tag title.
+	 *
+	 * @var string
+	 */
+	public $title;
+
+	/**
+	 * Tag description.
+	 *
+	 * @var string
+	 */
+	public $description;
+
+	/**
+	 * Tag callback.
+	 *
+	 * @var callable
+	 */
+	public $callback;
+
+	/**
+	 * Whether the tag is based on a segmentation criteria.
+	 *
+	 * @var bool
+	 */
+	public $use_criteria;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param string $name Tag name.
+	 * @param array  $args Tag arguments.
+	 */
+	public function __construct( $name, $args = [] ) {
+		$this->name         = $name;
+		$this->title        = $args['title'] ?? '';
+		$this->description  = $args['description'] ?? '';
+		$this->use_criteria = $args['use_criteria'] ?? false;
+		$this->callback     = $args['callback'] ?? '__return_empty_string';
+	}
+
+	/**
+	 * Get tag value.
+	 *
+	 * @return string
+	 */
+	public function get_value() {
```

Comment:
since this is a HTML output, and not just a value, maybe a more descriptive method name?

Reply:
Good call, renamed it in 206102ca175c47a07c14a897899e10f89a765a89